oracle/pkg/updater/updater.go

```diff
@@ -233,42 +175,65 @@ func (u *Updater) Start(ctx context.Context) <-chan struct{} {
 	return doneChan
 }
 
-func (u *Updater) handleEncryptedCommitment(
-	ctx context.Context,
-	update *preconf.PreconfmanagerUnopenedCommitmentStored,
-) error {
-	err := u.winnerRegister.AddEncryptedCommitment(
-		ctx,
-		update.CommitmentIndex[:],
-		update.Committer.Bytes(),
-		update.CommitmentDigest[:],
-		update.CommitmentSignature,
-		update.DispatchTimestamp,
-	)
-	if err != nil {
-		// ignore duplicate private key constraint
-		var pqErr *pq.Error
-		if errors.As(err, &pqErr) && pqErr.Code == "23505" {
-			u.logger.Warn(
-				"encrypted commitment already exists",
-				"commitmentIdx", common.Bytes2Hex(update.CommitmentIndex[:]),
-			)
-			return nil
+func gasUsedUntil(pos int, txns map[string]TxMetadata) uint64 {
+	var gasUsed uint64
+	for _, details := range txns {
+		if details.PosInBlock >= pos {
+			continue
 		}
-		u.logger.Error(
-			"failed to add encrypted commitment",
-			"commitmentIdx", common.Bytes2Hex(update.CommitmentIndex[:]),
-			"error", err,
-		)
-		return err
+		gasUsed += details.GasUsed
 	}
-	u.metrics.EncryptedCommitmentsCount.Inc()
-	u.logger.Debug(
-		"added encrypted commitment",
-		"commitmentIdx", common.Bytes2Hex(update.CommitmentIndex[:]),
-		"dispatch timestamp", update.DispatchTimestamp,
-	)
-	return nil
+	return gasUsed
+}
+
+func checkPositionConstraintSatisfied(
+	c *bidderapiv1.PositionConstraint,
+	txnDetails TxMetadata,
+	txns map[string]TxMetadata,
+) bool {
+	switch c.Basis {
+	case bidderapiv1.PositionConstraint_BASIS_ABSOLUTE:
+		switch c.Anchor {
+		case bidderapiv1.PositionConstraint_ANCHOR_TOP:
+			if txnDetails.PosInBlock <= int(c.Value) {
+				return true
+			}
+		case bidderapiv1.PositionConstraint_ANCHOR_BOTTOM:
+			if txnDetails.PosInBlock >= len(txns)-int(c.Value) {
+				return true
+			}
+		}
+	case bidderapiv1.PositionConstraint_BASIS_PERCENTILE:
+		switch c.Anchor {
+		case bidderapiv1.PositionConstraint_ANCHOR_TOP:
+			if txnDetails.PosInBlock <= (len(txns)*int(c.Value))/100 {
+				return true
+			}
+		case bidderapiv1.PositionConstraint_ANCHOR_BOTTOM:
+			if txnDetails.PosInBlock >= (len(txns)*(100-int(c.Value)))/100 {
+				return true
+			}
+		}
+	case bidderapiv1.PositionConstraint_BASIS_GAS_PERCENTILE:
+		gasUsed := gasUsedUntil(txnDetails.PosInBlock, txns)
+		gasPercentile := (gasUsed * 100) / txnDetails.TotalGas
+		txnGasPercentile := (txnDetails.GasUsed * 100) / txnDetails.TotalGas
+		if txnGasPercentile > uint64(c.Value) {
```

Comment:
I don't think this is always correct: Assume `c.Value = 20`, `txnGasPercentile = 10`, and the transaction comes after 50% of the total block gas. I'd say it should not count as satisfied because it is entirely outside of the requested range.